### PR TITLE
ddt removal in ui template tests

### PR DESCRIPTION
**Pull request by sthirugn**
Test results:
```sh
# nosetests tests/foreman/ui/test_template.py

----------------------------------------------------------------------
Ran 11 tests in 256.942s

OK
```

### CODE REVIEW

File: tests/foreman/ui/test_template.py [hunk 1]
```diff
@@ -53,36 +56,14 @@ def test_negative_create_template_with_too_long_name(self):
...
-            make_templates(
-                session,
-                name=name,
-                template_path=get_data_file(OS_TEMPLATE_DATA_FILE),
-                custom_really=True,
-                template_type='provision',
-            )
-            self.assertIsNotNone(self.template.wait_until_element
-                                 (common_locators['name_haserror']))
+            for name in invalid_values_list():
```
Comment (oshtaier) [inline tests/foreman/ui/test_template.py]: `for name in invalid_values_list(interface='ui') `?

File: tests/foreman/ui/test_template.py [hunk 2]
```diff
@@ -186,18 +156,15 @@ def test_positive_create_snippet_template(self, name):
...
+                with self.subTest(name):
+                    make_templates(session, name=name,
+                                   template_path=SNIPPET_DATA_FILE,
+                                   custom_really=True, snippet=True)
+                    self.assertIsNotNone(self.template.search(name))
 
-    @data(*generate_strings_list(len1=8))
     @run_only_on('sat')
-    def test_remove_template(self, template_name):
+    def test_remove_template(self, ):
```
Comment (oshtaier) [inline tests/foreman/ui/test_template.py]: `def test_remove_template(self):` ?

**Comment by oshtaier** [comment]
I am not sure that your format transformation are okay. E.g:
```
make_templates(
-                    session,
-                    name=name,
-                    template_path='',
-                    custom_really=True,
-                    template_type='PXELinux',
-                )
+                make_templates(session, name=name, template_path='',
+                               custom_really=True, template_type='PXELinux')
```
Actually, I spent some time before to transform formatting that used by you on format that was before that PR

**Comment by oshtaier** [comment]
pending questions/comments

File: tests/foreman/ui/test_template.py (hunk 1 above)
Comment (sthirugn) [inline tests/foreman/ui/test_template.py]: :+1:

**Comment by sthirugn** [comment]
```py
-               make_templates(
-                    session,
-                    name=name,
-                    template_path='',
-                    custom_really=True,
-                    template_type='PXELinux',
-                )
+                make_templates(session, name=name, template_path='',
+                               custom_really=True, template_type='PXELinux')
```
As per pep8 standards:
- the first solution (which you used) is the hanging indent in which the opening parenthesis of a parenthesized statement is the last non-whitespace character of the line, with subsequent lines being indented until the closing parenthesis.
- the second solution (I used) is the alignment with opening delimiter.  According to pep8 example:
```py
# Aligned with opening delimiter.
foo = long_function_name(var_one, var_two,
                         var_three, var_four)
```
Both are legal as per pep8 but the second solution is 2 lines vs. 7 lines in first solution.  So I preferred the second option.

**Comment by sthirugn** [comment]
Comments addressed

**Comment by oshtaier** [comment]
ACK

**Comment by abalakh** [comment]
Just commented about code style in #2927 and it appears that you've already discussed it here. OK, both options are valid according to PEP8. But, according to PEP8, dozen of styles are valid and it's the matter of project, which style to follow. For robottelo we were always trying to avoid using alignment with opening delimiter and prefer using regular intents (you can see @Ichimonji10 comments in #2077, for example) for a couple of reasons like better readability, clearer diffs in case of any arguments change etc, it doesn't actually matter now. What matters is the project is following this style, and you're introducing another one. IMO you can either:
a) follow the regular intents style to be consistent with everyone else
b) update _all_ the continuation lines according to proposed style and kindly ask everybody to follow the style.
Otherwise, if everyone starts using everything what is not forbidden by PEP8, our code will become a real mess of styles and approaches.

**Comment by abalakh** [comment]
Pending question

**Comment by sthirugn** [comment]
Two things here:
* @abalakh I agree to your comments in https://github.com/SatelliteQE/robottelo/pull/2927 and it is fixed.  
* I think my approach in https://github.com/SatelliteQE/robottelo/pull/2928#issuecomment-150592631will reduce the number of lines greatly.  I took this approach only for the UI since the UI functions take a lot of arguments (many times illegal as per pylint and other tools).  Can you explain what your concern here is?

**Comment by sthirugn** [comment]
@abalakh and I had a chat in irc regarding this, I will fix this PR.

**Comment by sthirugn** [comment]
All comments addressed and rebased.

**Comment by abalakh** [comment]
ACK

**Comment by sthirugn** [comment]
part of https://github.com/SatelliteQE/robottelo/issues/2769

**Pull request merged**